src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -105,15 +112,9 @@ public KafkaTopicRepository(final ZooKeeperHolder zkFactory, final KafkaFactory
     public void createTopic(final EventType eventType) throws TopicCreationException, DuplicatedEventTypeNameException {
         createTopic(eventType.getTopic(),
                 calculateKafkaPartitionCount(eventType.getDefaultStatistic()),
-                settings.getDefaultTopicReplicaFactor(),
-                getTopicRetentionMs(eventType),
-                settings.getDefaultTopicRotationMs());
-    }
-
-    private long getTopicRetentionMs(final EventType eventType) {
-        return Optional.ofNullable(eventType.getOptions())
-                .map(EventTypeOptions::getRetentionTime)
-                .orElse(settings.getDefaultTopicRetentionMs());
+                nakadiSettings.getDefaultTopicReplicaFactor(),
+                eventType.getOptions().getRetentionTime(),
```

Comment:
You should check retentionTime for null here as it is nullable. I understand that you expect it not to be null when this method is called, but as we declare it nullable - we should check it for null. If it is null - I think we should throw exception and Nakadi should return 500 (because something really went wrong if we have it null here)